### PR TITLE
DEPR: is_period_dtype, is_sparse

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

is_sparse is particularly non-performant bc it uses a runtime import

### CODE REVIEW

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -5226,6 +5226,7 @@ See the `Full Documentation <https://github.com/wesm/feather>`__.
 Write to a feather file.
 
 .. ipython:: python
+   :okwarning:
```
Comment (phofl) [inline doc/source/user_guide/io.rst:5229]: What warning are these examples showing?

Comment (jbrockmendel) [reply doc/source/user_guide/io.rst:5229]: i dont remember off the top of my head, just that it was one of the is_foo checks

Comment (mroeschke) [reply doc/source/user_guide/io.rst:5229]: Does it mean pyarrow are using these methods then?

Comment (jbrockmendel) [reply doc/source/user_guide/io.rst:5229]: Looks like pyarrow/pandas_compat.py:373 is using is_sparse

Comment (mroeschke) [reply doc/source/user_guide/io.rst:5229]: Okay I think it's reasonable for them to use the isinstance check

Comment (jbrockmendel) [reply doc/source/user_guide/io.rst:5229]: gentle ping

**Review by mroeschke — APPROVED** [review]
LGTM. Merge when ready @phofl

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @jbrockmendel

**Pull request merged**